### ISSUE TITLE
Asset Meta Users crashes CP for logged in user

### ISSUE THREAD

**Issue by Rovaski** [statamic/cms#2617]
## Bug Description
Say you have an asset meta field (in our case called 'uploaded by') that stores the user that uploads the asset. 
Now, say that that same uploader, uses that asset as their avatar. 
That user will now no longer be able to access the CP until either someone un-sets that users avatar, or removes that user from the 'uploaded by' field in the asset. 

## How to Reproduce
Fresh Install or Not, doesn't matter.
Set the blueprint of Assets to have a user field if you don't already. In my case, this is called Uploader or Uploaded By
Set that field in an asset to be the same user that is currently logged in, e.g. Set Uploaded By to be the same user logged in.
Now set that asset you modified, to be the avatar image for the logged in user. 
Experience the fun. 

## Extra Detail
Blueprint: assets.yaml
```
sections:
  main:
    display: Main
    fields:
      -
        handle: alt
        field:
          display: 'Alt Text'
          type: text
          instructions: 'Description of the image'
      -
        handle: uploaded_by
        field:
          max_items: 1
          mode: select
          display: 'Uploaded By'
          type: users
          icon: users
          listable: hidden
```

Meta from an asset:
```
data:
  uploaded_by: a4dda7a6-0789-413e-9c17-7a480131e4b0
size: 8054
last_modified: 1602169187
width: 256
height: 256
```
Note. the uploaded_by user id is the same as the current logged in user. It doesn't 'crash' if the uploader isn't the currently logged in user. 

## Environment
Statamic 3.0.16 Pro
Laravel 7.28.4
PHP 7.4.6
No addons installed

- Fresh install from `statamic/statamic`

**Comment by Rovaski** [comment]
Still an issue in 3.0.37.

**Comment by duncanmcclean** [comment]
Confirmed. I think the issue here is that because you have the user set on the asset and the asset set on the user, it's creating an infinite loop between the two.

If you remove the user field from the asset (or vice versa), it should resolve your issue for now.

**Comment by duncanmcclean** [comment]
I think this may have already been fixed by #6163, it's no longer happening for me (a similar thing was happening at one point for entry fields which is also resolved).
